sound/soc/sof/intel/hda.c

```diff
@@ -597,7 +597,7 @@ int hda_dsp_probe(struct snd_sof_dev *sdev)
 		 * not all errors are due to memory issues, but trying
 		 * to free everything does not harm
 		 */
-		goto err;
+		goto free_streams;
 	}
```

Comment:
initial streams fails, no streams to be freed? if streams are partially initialized success, better to do free to those initialized inside stream_init(), at e.g. it's exiting of function. 
should be "goto dsp_bar_unmap;"